test/unit/migrations/050-test.js

```diff
@@ -0,0 +1,81 @@
+import { strict as assert } from 'assert'
+import sinon from 'sinon'
+import migration50 from '../../../app/scripts/migrations/050'
+
+const LEGACY_LOCAL_STORAGE_KEYS = [
+  'METASWAP_GAS_PRICE_ESTIMATES_LAST_RETRIEVED',
+  'METASWAP_GAS_PRICE_ESTIMATES',
+  'cachedFetch',
+  'BASIC_PRICE_ESTIMATES_LAST_RETRIEVED',
+  'BASIC_PRICE_ESTIMATES',
+  'BASIC_GAS_AND_TIME_API_ESTIMATES',
+  'BASIC_GAS_AND_TIME_API_ESTIMATES_LAST_RETRIEVED',
+  'GAS_API_ESTIMATES_LAST_RETRIEVED',
+  'GAS_API_ESTIMATES',
+]
+
+describe('migration #50', function () {
+  const mockLocalStorageRemoveItem = sinon.stub()
+
+  beforeEach(function () {
+    window.localStorage.removeItem = mockLocalStorageRemoveItem
+  })
+
+  afterEach(function () {
+    sinon.restore()
+  })
```

Comment:
I don't think this works. `removeItem` never gets restored, as the reference to the original `removeItem` is lost.

In order to stub one method of an object, you need to use the form `sinon.stub(baseObject, 'methodName'). Then `sinon` keeps a reference to the original and restores it when you call `sinon.restore`.

Also, you're sharing a mock between all tests here as well, which is not good.

I think something like this would work? (not 100% sure - didn't test this)

```suggestion
  let mockLocalStorageRemoveItem

  beforeEach(function () {
    mockLocalStorageRemoveItem = sinon.stub(window.localStorage, 'removeItem')
  })

  afterEach(function () {
    sinon.restore()
  })
```

Reply:
@Gudahtt I applied this fix in fbc1a7ac4

Successfully passing the tests also required adding a `remove` item method to the window.localstorage mock property in the tests, as the migration gets run in another unit test too.